### DISCUSSION TITLE
btrack particle ID consistency across runs?

### DISCUSSION THREAD

**Discussion by thejasvibr** [quantumjot/btrack#297] [Q&A] [answered]
Hi, 
I'm currently using ```btrack``` to track my bats flying around in a cave, and it's doing really well in comparison to the other packages I've tried :) (definitely an edge use case I'm aware).

There are a few points where the tracking isn't so reliable, and I'd like to go back and correct the tracking data manually by entering new rows in the napari format numpy array. 

Given that the mask images + ```btrack``` parameters are the same each time tracking is done - will it always result in the *same* particle ID number? If yes, then it'd be great to hear a confirmation. If not, is there a way to ensure consistent particle ID numbers? My end goal is to get 3D trajectories after 2D tracking from multiple cameras - and here having a way to get consistent particle ID numbers would be great.

**Comment by quantumjot** [comment, marked as answer]
Hi @thejasvibr - this sounds super cool! I'm glad btrack is working well for this - I'd love to see a video at some point.

Re: track ID - yes, it should be the same between runs on the same dataset if everything is done the same way. However, it most likely depends on the order of adding objects to the tracker using the `append` method. With different datasets (e.g. different 2D views in your case), there is no guarantee that the same object will have a matching track ID between the different runs/datasets.

**Reply by thejasvibr** [reply]
Awesome,thanks - will def send you a link to a video with tracking once I have something presentable :)
